ui/app/src/hooks/useEnhancedDialogState.ts

```diff
@@ -75,5 +85,5 @@ export function useEnhancedDialogState(initialState?: Partial<EnhancedDialogStat
       onHasPendingChange,
       onSubmittingAndOrPendingChange
     };
-  }, [setState, state]);
+  }, [initialState, setState, state]);
```

Comment:
This is dangerous to include here as users of the effect most likely won’t memo the initial state. Ignore it and add a comment to justify the eslint ignore